Improvements for failed imports (#4845)
Particular/docs.particular.net: servicecontrol/import-failed-messages.md

NOTE: Messages that have corrupt (i.e. unreadable, not deserializable) header data will fail to be process at all and be moved to the 'error' queue (by default `particular.servicecontrol.error` or `particular.servicecontrol.audit.error`)

Messages that fail to be imported are stored in the ServiceControl database in the `FailedAuditImports` and `FailedErrorImports` collections.

In addition, a log with the failure reason is written for the message in the [`%ServiceControl/LogPath%`](/servicecontrol/creating-config-file.md#host-settings-servicecontrollogpath)`\FailedImports\{Audit|Error}\%failureid%.txt`. These messages will not be visible in ServiceInsight.

## Failed message custom check

When a message that has failed the import is detected in the ServiceControl database, the [**Message Ingestion** custom check](/servicecontrol/servicecontrol-instances/#self-monitoring-via-custom-checks-failed-imports) is marked as failed to bring the failed import(s) to the administrator's attention.

## How to reimport

> To reimport the failed messages, the instance to be shut down and started from a command line

Reviewer: ```suggestion
To reimport the failed messages, the instance must be shut down and started from a command line using one of the following commands:
```

NOTE: The value to use for `--serviceName` is equal to the instance name. It also is available in the Windows Service information

> ServiceControl instance:

Reviewer: ```suggestion
**ServiceControl instance:**
```

> Using the additional argument `--import-failed-errors`.

Reviewer: ```suggestion
```

```cmd
ServiceControl.exe --serviceName=Particular.Servicecontrol --import-failed-errors
```

> ServiceControl audit instance:

Reviewer: ```suggestion
**ServiceControl audit instance:**
```

> Using the additional argument `--import-failed-audits`

Reviewer: ```suggestion
```

```cmd
ServiceControl.Audit.exe --serviceName=Particular.Servicecontrol.Audit --import-failed-audits
```

> While in import mode, ServiceControl or ServiceControl Audit will not be processing its input queues. Once the imported failure message is re-processed successfully, the message will be available in ServicePulse and ServiceInsight.

Reviewer: ```suggestion
While in import mode, ServiceControl or ServiceControl Audit will not be processing its input queues. Once the message is re-processed successfully, it will be available in ServicePulse and ServiceInsight. ServiceControl or ServiceControl Audit instance can then be started normally again.
```

> The custom check will disappear if no failed imports remain.

Reviewer: ```suggestion
The custom check will no longer be displayed if all failed imports have been successfully reimported
```
